### ISSUE TITLE
During data upload line geometry is converted to point geometry

### ISSUE THREAD

**Issue by blewis** [cga-harvard/CAMP#6]

**Comment by capooti** [comment]
There is a problem with style creation in GeoServer, I am looking into it and I created a ticket here:
https://github.com/GeoNode/geonode/issues/4827

**Comment by capooti** [comment]
I have just deployed a PR to GeoNode that addresses the issue:
https://github.com/GeoNode/geonode/pull/4837/files

The PR is already on the staging, so you can test it now. Let me know if I can close this @wendy-guan  @blewis

**Comment by wendy-guan** [comment]
I just told Prof. Xu to test this on staging.

**Comment by capooti** [comment]
Great. Remember, source code of the PR is still not merged in GeoNode. They need to wait until tomorrow for using it in their staging

**Comment by wendy-guan** [comment]
I only asked him to test on our staging. He just replied saying that he has tried and it worked for him as expected. Thanks!

**Comment by capooti** [comment]
good :)

**Comment by wendy-guan** [comment]
Prof. Xu said the ZJU production server can't roll back for whatever reason, therefore urgently need this fix. Please make it available to Wang Lijun ASAP. Thanks!

**Comment by wendy-guan** [comment]
Uploaded line features are showing up corrected as lines in a map now, but the automatically generated thumbnail still shows features as points.

**Comment by capooti** [comment]
@wendy-guan I can't replicate this in CAMP staging

**Comment by wendy-guan** [comment]
I will ask Prof. Xu to send a video and the layer - if he still experience the problem now.

**Comment by wendy-guan** [comment]
video attached. If you search for "fengxuefeng" you will find several line layers with the thumbnail as points. They were uploaded from shapefiles.
[20190918_203024.zip](https://github.com/cga-harvard/CAMP/files/3626673/20190918_203024.zip)

**Comment by capooti** [comment]
@wendy-guan it would be interesting to know if this is still happening. On staging this looks fixed after last GeoNode updates

**Comment by wendy-guan** [comment]
"fixed after last GeoNode updates" when was it?

**Comment by wendy-guan** [comment]
You may see the time stamp of the layers with this problem in staging now. 
http://128.31.22.46/search/?limit=20&offset=0
http://128.31.22.46/layers/geonode:line_of_fenxuefeng_1#info

**Comment by wendy-guan** [comment]
Please note that the problem now is:  Uploaded line features are showing up corrected as lines in a map now, but the automatically generated thumbnail still shows features as points.

**Comment by capooti** [comment]
oh I see, yes there is still an issue with thumbnails. Will create a ticket in GeoNode.
Luckily at least this can be mitigated by uploading a thumbnail for now.

**Comment by wendy-guan** [comment]
Do you mean that you will not fix this, but wait for someone in the GeoNode community to fix it - which we don't know when?

**Comment by capooti** [comment]
@wendy-guan I am planning to fix it but meanwhile I do that maybe someone else can take care of it

**Comment by wendy-guan** [comment]
OK, That makes sense. I will let Prof. Xu know. Thanks!

**Comment by capooti** [comment]
Reopening this, I am looking again into this GeoNode bug:
https://github.com/GeoNode/geonode/issues/4827

will keep you posted.

**Comment by capooti** [comment]
I debugged the problem and it looks to be a GeoServer scalability problem. I have commented my investigation in the above GeoNode ticket. Let's wait for a response from some GeoServer expert.

**Comment by wendy-guan** [comment]
Any update on this one?  Really hope we can close all issues soon and move on...

**Comment by capooti** [comment]
we are really close to fix this as well @wendy-guan 👍 a PR is being tested
I am planning to try this Tuesday as well and confirm you we are done with it too

**Comment by capooti** [comment]
@wendy-guan @blewis I just submitted a PR in GeoNode which should be able to definitely fix this:
https://github.com/GeoNode/geonode/pull/5024

This PR is already in CAMP staging for you to test. I will let you know when it is part of GeoNode master, so that CAMP staff can update their instance.

**Comment by wendy-guan** [comment]
Prof. Xu tested on staging last night and found line layers are still generating point thumbnails. Examples are "route" and "line_of_fenxuefeng" layers.

**Comment by capooti** [comment]
hi @wendy-guan this is because the PR I submitted is still not in code base. I am waiting for feedback from other developers. Meanwhile I could fork GeoNode, if this is urgent, and they can use it until PR is not merged. What do you think?

**Comment by capooti** [comment]
this is the PR to keep in mind: https://github.com/GeoNode/geonode/pull/5024

**Comment by capooti** [comment]
oh @wendy-guan you mean in the staging. Sorry, I will check it right now.

**Comment by capooti** [comment]
yes @wendy-guan this is expected, as staging was using master as well. I reverted now to use the PR source code. In any case, could Prof Xu upload the thumbnail when this is happening until the fix is included in GeoNode source code?

**Comment by wendy-guan** [comment]
3 days ago
@wendy-guan @blewis I just submitted a PR in GeoNode which should be able to definitely fix this:GeoNode/geonode#5024 This PR is already in CAMP staging for you to test. I will let you know when it is part of GeoNode master, so that CAMP staff can update their instance.
--
I might have misunderstood the above message - as to what was fixed by the PR where and when. Anyway, just let me know when this bug is fixed on staging so I can ask Prof. Xu to test again.

**Comment by capooti** [comment]
I believe that the bug in the subject of this ticket is now fixed on staging (and still not in GeoNode, we need still to merge the code).
I start thinking that the thumanail issue is generated by another (similar) problem in a different part of GeoNode. Will look into this on Tuesday and let you know.

**Comment by wendy-guan** [comment]
Prof. Xu tested again on staging, and the result is inconsistent, some line layers generated line thumbnails, others still generated point thumbnails. See attached.
![Snap2](https://user-images.githubusercontent.com/8332594/66759432-413f6e00-ee6e-11e9-8d8a-a1e6ac4bb78f.jpg)

**Comment by capooti** [comment]
Yes, @wendy-guan  I am aware of the problems with thumbnail generation. I will work on a fix tomorrow

**Comment by capooti** [comment]
this is now fixed in staging.

**Comment by wendy-guan** [comment]
Test on staging is positive. Is the fix available for ZJU server now?

**Comment by capooti** [comment]
hi @wendy-guan I was still unable to merge the PR as there is one test which is failing - I am currently working to check out why this is happening today. I will keep you updated on this process and let you know when it is available

**Comment by capooti** [comment]
just sent to the CAMP staff the link to a GeoNode branch containing the fixes for this issue
https://github.com/capooti/geonode/tree/camp-before-2.10.1-merge

**Comment by blewis** [comment]
I just tested on Staging with this line layer http://128.31.22.46/layers/geonode:route_2_3 and the line layer was converted to points and the thumbnail is points.  NOTE clicking on a point however returns a highlighted LINE via WFS.

**Comment by capooti** [comment]
thanks @bglewis I have sent a new fix here:
https://github.com/GeoNode/geonode/issues/4827

you can already test it in CAMP staging, will let you know when it is merged in 2.10.1

**Comment by capooti** [comment]
the PR was merged in GeoNode and will be part of GeoNode 2.10.2 which will be released shortly.
Meanwhile CAMP staff can use the above branch - which is also installed in CAMP staging

**Comment by wendy-guan** [comment]
What is the status of this now? Has it passed the test on CAMP staging? Has the fix been part of GeoNode release available to AMAP production?

**Comment by capooti** [comment]
@wendy-guan the changes are included in latest geonode - I just updated CAMP.

GeoNode 2.10.2 should be released in days, I will let you know when it is available.
But until then it is safe to use GeoNode 2.10.x branch - that code is already used by a number of GeoNode instances worldwide:

https://github.com/GeoNode/geonode/tree/2.10.x
